query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala

```diff
@@ -130,7 +130,7 @@ object RowAggregator {
           + s"or calculate the rate and histogram_quantile before applying the aggregation. "
           + s"If you have a genuine use case for this query, please get in touch.")
       }
-    } else if (valColType == ColumnType.DoubleColumn) {
+    } else if (valColType == ColumnType.DoubleColumn || valColType == ColumnType.StringColumn) {
```

Comment:
Why do we need `else if`?  Original intent of the PR is to handle histograms which is already done just let everything else follow the same path as before. By restricting the types we don't know additional regression we may see.

Reply:
good point. The patch removed the default case so that compile would fail if missing any cases.

Reply:
@vishramachandran , requesting your feedback.

Reply:
The current patch doesn't work for topk/bottomk as Im getting 
``topk(2, foo{..})`` fails with `The TopK operation is not supported for StringColumn`
thus I strongly feel as I originally recommended, minimize the blast radius and restrict your changes to only histograms, which was the original intention. Everything else should take the same code path as before.

Reply:
you are right.

I think the problem may be caused by this function
`  def valueColumnType(schema: ResultSchema): ColumnType = {
    require(schema.isTimeSeries, s"Schema $schema is not time series based, cannot continue query")
    require(schema.columns.size >= 2, s"Schema $schema has less than 2 columns, cannot continue query")
    schema.columns(1).colType
  }
`
For schema more than 2 columns schema.columns(1).colType may be always be the valueColumnType